sympy/liealgebras/type_e.py

```diff
@@ -49,32 +49,45 @@ def simple_root(self, i):
 
         This method returns the ith simple root for E_n.
 
+        Sources
+        =======
+        - https://en.wikipedia.org/wiki/E8_(mathematics)#Simple_roots
+        - https://en.wikipedia.org/wiki/E7_(mathematics)#Root_system
+        - https://en.wikipedia.org/wiki/E6_(mathematics)#Roots_of_E6
+
         Examples
         ========
 
         >>> from sympy.liealgebras.cartan_type import CartanType
         >>> c = CartanType("E6")
         >>> c.simple_root(2)
-        [1, 1, 0, 0, 0, 0, 0, 0]
+        [0, 1, -1, 0, 0, 0]
+
         """
         n = self.n
-        if i == 1:
-            root = [-0.5]*8
-            root[0] = 0.5
-            root[7] = 0.5
-            return root
-        elif i == 2:
-            root = [0]*8
-            root[1] = 1
-            root[0] = 1
-            return root
-        else:
-            if i == 7 or i == 8 and n == 6:
-                raise ValueError("E6 only has six simple roots!")
-            if i == 8 and n == 7:
-                raise ValueError("E7 has only 7 simple roots!")
-
-            return self.basic_root(i-3, i-2)
+        def _dim8_weights(i,n):
+            """
+            Returns the roots in a basis
+            that makes it easy to subset
+            e7 and e6.
+
+            """
+            e = eye(8)
+            if i == n:
+                return (e.row(n-3) - e.row(n-2)).T
+            if i == n - 1:
+                return  - ones(8,1) / 2
+            if i == n - 2:
+                return (e.row(n-3) + e.row(n-2)).T
+            else:
+                return (e.row(i-1) - e.row(i)).T
+
+        _root = _dim8_weights(i, n).T
+        if i != 8:
+            _root[n-1] = sqrt(sum(_root[j]**2 for j in range(n-1, 8)))
```

Comment:
Where does this come from?

Reply:
This is coming from the fact you can subgroup E6 and E7 from E8.  Here is the equation in Wikipedia https://en.wikipedia.org/wiki/E6_(mathematics)#E6_roots_derived_from_the_roots_of_E8 but I'm looking for another "not from wikipedia" source

Reply:
Here is a wikipedia article https://en.wikipedia.org/wiki/E6_(mathematics)#E6_roots_derived_from_the_roots_of_E8 but I'm looking for a non wiki source